cdap-master/src/main/java/co/cask/cdap/data/tools/HBaseTableExporter.java

```diff
@@ -0,0 +1,222 @@
+
+  public HBaseTableExporter() throws Exception {
+    this.hConf = HBaseConfiguration.create();
+    this.cConf = CConfiguration.create();
+    Injector injector = createInjector(cConf, hConf);
+    this.txClient = injector.getInstance(TransactionSystemClient.class);
+    this.txService = injector.getInstance(TransactionService.class);
+    this.zkClientService = injector.getInstance(ZKClientService.class);
+
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      @Override
+      public void run() {
+        try {
+          HBaseTableExporter.this.stop();
+        } catch (Throwable e) {
+          LOG.error("Failed to stop the tool.", e);
+        }
+      }
+    });
+
+  }
+
+  private static Injector createInjector(CConfiguration cConf, Configuration hConf) {
+    return Guice.createInjector(
+      new ConfigModule(cConf, hConf),
+      new IOModule(),
+      new ZKClientModule(),
+      new KafkaClientModule(),
+      new LocationRuntimeModule().getDistributedModules(),
+      new DiscoveryRuntimeModule().getDistributedModules(),
+      new DataFabricModules().getDistributedModules(),
+      new DataSetsModules().getDistributedModules(),
+      new MetricsClientRuntimeModule().getDistributedModules(),
+      new LoggingModules().getDistributedModules(),
+      new ExploreClientModule(),
+      new ViewAdminModules().getDistributedModules(),
+      new StreamAdminModules().getDistributedModules(),
+      new NotificationFeedServiceRuntimeModule().getDistributedModules()
+    );
+  }
+
+  /**
+   * Sets up the actual MapReduce job.
+   * @param tx The transaction which needs to be passed to the Scan instance. This transaction is be used by
+   *           coprocessors to filter out the data corresonding to the invalid transactions .
+   * @param tableName Name of the table which need to be exported as HFiles.
+   * @return the configured job
+   * @throws IOException
+   */
+  public Job createSubmittableJob(Transaction tx, String tableName) throws IOException {
+
+    Job job = Job.getInstance(hConf, "HBaseTableExporter");
+
+    job.setJarByClass(HBaseTableExporter.class);
+    Scan scan = new Scan();
+    scan.setCacheBlocks(false);
+    // Set the transaction attribute for the scan.
+    scan.setAttribute(TxConstants.TX_OPERATION_ATTRIBUTE_KEY, new TransactionCodec().encode(tx));
+    job.setNumReduceTasks(0);
+
+    TableMapReduceUtil.initTableMapperJob(tableName, scan, Import.KeyValueImporter.class, null, null, job);
+
+    FileSystem fs = FileSystem.get(hConf);
+    Random rand = new Random();
+    Path root = new Path(fs.getWorkingDirectory(), "hbasetableexporter");
+    fs.mkdirs(root);
+    while (true) {
+      bulkloadDir = new Path(root, "" + rand.nextLong());
+      if (!fs.exists(bulkloadDir)) {
+        break;
+      }
+    }
+
+    HFileOutputFormat2.setOutputPath(job, bulkloadDir);
+    HTable hTable = new HTable(hConf, tableName);
+    HFileOutputFormat2.configureIncrementalLoad(job, hTable);
```

Comment:
Validated on the HBase 1.0.0-cdh5.4.4 cluster. Source table had 4 regions, while the destination table was created with only one region. I was able to migrate the HBase table data correctly. All the data went into the single region on the destination table.

However I agree that its better to pre-split the destination table as per the same keys and number of regions as the source. I have filed JIRA for this https://issues.cask.co/browse/CDAP-4350